docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+````
+_________________________________
+Welcome to scheduler--;!
+What can I do for you?
+_________________________________
+File Created: Personal
+File Created: Zoom
+File Created: Timetable
+File Created: Goal
+The file has successfully been loaded!
+_________________________________
+You have no events today!
+_________________________________
+````
+## Design
+### Architecture
 
+![Architecture Diagram](./diagrams/Architecture.jpg)
 
-## Product scope
-### Target user profile
+The ***Architecture Diagram*** given above explains the high-level design of App. Given below is a quick overview of
+each component.
 
-{Describe the target user profile}
+`Duke` class contains the main method. It is responsible for,
 
-### Value proposition
+- At app launch: Initializes the components in the correct sequence, and connects them up with each other.  
+- At shut down: Stores the user data from memory into the hard disk.
 
-{Describe the value proposition: what problem does it solve?}
+The rest of the App consists of five components.
 
-## User Stories
+- `Ui`: The Ui of the App.
+- `Parser`: Parses the input to determine which command to execute.
+- `Command`: The command executer.
+- `UserData`: Holds the data of the App in the memory.
+- `Storage`: Reads data from, and writes data to, the hard disk.
+
+#### Ui Component
+The 'Ui' component is in charge of handling input from users and system output.
+
+#### Command Component
+
+![Diagram for commmand](./diagrams/Command.jpg)
+
+API: [Command.java](https://github.com/AY2021S1-CS2113T-T12-4/tp/blob/master/src/main/java/seedu/duke/command/Command.java) 
+1. `Parser` class to parse the user command 
+1.  This results in a `Command` object executed by the `Parser`
+1.  The command execution can affect the `Model`
+
+It listens for commands made in the Duke Class and sends the input to the parser class.
+It is also responsible for printing messages from commands and exception messages. 
+
+#### Parser Component
+![Parser diagram](./diagrams/Parser_diagram.jpg)
+
+The `Parser` object uses its `parse` method to parse inputs passed to it from `Ui`. This method returns a `Command` object to be executed by `Duke`.
+
+#### UserData Component
+![UserData diagram](./diagrams/UserData_diagram.jpg)
+
+The `UserData`
+- stores `EventList` objects that store events of different categories (i.e. Personal, Timetable, Zoom).
+- stores a `Goal` object that represents the current user goal.
+- does not depend on the other components.
+
+#### Storage Component
+![Diagram for storage class](./diagrams/classDiagramStorage.jpg)
+
+API [Storage.java](https://github.com/AY2021S1-CS2113T-T12-4/tp/blob/master/src/main/java/seedu/duke/storage/Storage.java) 
+
+The storage component,
+- is able to store all event and goal information onto the computer
+- is able to load all event and goal information from the computer into the program
+
+![Diagram for storageOverall](./diagrams/storageOverall.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/60382285/97659226-69196800-1aa9-11eb-964c-cb18d25f0743.png)
In addition to the above comment, you can consider finding a common place to label the arrows, ie label all below / on top of the arrow as the current diagram has labels on top of and on the arrows itself. This would make the diagram look neater for the reader.

Also it is not immediately clear which you are referring to, loading or saving. Perhaps putting the loading or saving on top in the middle would help with this as putting it in the top left makes it easier to miss